Sources/NIOHTTPClient/ConnectionPool.swift

```diff
@@ -0,0 +1,376 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIOHTTPClient open source project
+//
+// Copyright (c) 2018-2019 Swift Server Working Group and the SwiftNIOHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIOHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
```

Comment:
@adtrevor do we need the `Foundation` dependency? If it's just for the lock, you can just `import NIOConcurrencyHelpers`

Reply:
Yes because we need the `URL` type! 🙂

Reply:
@adtrevor ah right. That's an interesting case because using the `URL` type is clearly the right choice. However using anything from `Foundation` comes with the drawbacks that static linking won't work (because `Foundation` links libcurl, libxml2, and libssl). For now let's just ignore it and continue with the Foundation dependency. We can always make a `NIOHTTPClientCore` module that contains everything (and uses `String` instead of `URL`) and a `NIOHTTPClient` module which does `@_exported import NIOHTTPClientCore`. That way almost all users would just use `NIOHTTPClient` and get the `Foundation` dependency but if you really need to not have `Foundation` you can still `import NIOHTTPClientCore`. But for now, let's just ignore this all.

SwiftMetrics pulls off a [similar trick](https://github.com/apple/swift-metrics/tree/master/Sources)

Reply:
Interesting! I didn't know that Foundation caused this. Marking this as resolved for now, should we open an issue for this?

Reply:
Let’s resolve this